Automattic/WP-Job-Manager
# Introduce WP admin notices and refactor for setup wizard

Fixes #1181 (in a very long winded way 😃)

#### Changes proposed in this Pull Request:

* Primarily: Introduces admin notices helper for introducing notices in admin dashboard. 
* Removes the old way of redirecting to setup page on first activation. I never liked this as it made it sometimes made it difficult to access for bulk activations or if something interrupts activation. 
* Refactors setup class by:
  * Moving HTML to views.
  * Moving processing logic out of `output()` method.

#### Testing instructions:

* In an existing WPJM instance, switch to this branch. Deactivate and then reactivate the plugin. Nothing should happen and you shouldn't be prompted to set up WPJM.
* I a new instance, activate the plugin for the first time. You shouldn't be redirected to the setup wizard but on the dashboard, plugin, and job listing pages you should have a notice to set up WPJM.

#### To do
- [x] Add tests for `WP_Job_Manager_Admin_Notices` class.
